src/python/dxpy/scripts/dx.py

```diff
@@ -6715,14 +6715,14 @@ def __call__(self, parser, namespace, values, option_string=None):
     "--filter-json",
     "-j",
     type=str,
-    help='The full input JSON object as a string and corresponding to "--retrieve-expression". Must be used with "--retrieve-expression" flag. Either "--filter-json" or "--filter-json-file" may be supplied, not both.',
+    help='The full input JSON object as a string and corresponding to "--retrieve-expression". Must be used with "--retrieve-expression" flag. Either "--filter-json" or "--filter-json-file" may be supplied, not both. If no filter is provided, an SQL query returning all data will be generated. In this case, the --sql parameter must be used.',
```

Comment:
`If no filter is provided, an SQL ...`
to
`If no filter is provided, a SQL ...`